src/plugins/irc-events/message.js

```diff
@@ -67,11 +67,13 @@ module.exports = function(irc, network) {
 						type: Chan.Type.QUERY,
 						name: target,
 					});
-					network.channels.push(chan);
+
 					client.emit("join", {
 						network: network.id,
 						chan: chan.getFilteredClone(true),
+						index: network.addChannel(chan),
 					});
+					client.save();
```

Comment:
Why `client.save();` in some files and not in others?

Reply:
We only save when there's something to change. We dont' save in banlist and list because those aren't saved to the user file, so there's no need to save at that point.